### ISSUE TITLE
Pulling RID of a Vertex from Java JDBC Library

### ISSUE THREAD

**Issue by decafmainline** [orientechnologies/orientdb#6766]
## Operating System
- [ ] Linux
- [x] MacOSX
- [ ] Windows
- [ ] Other Unix
- [ ] Other, name? 

I just upgraded from 2.1.x to 2.2.10

I have a query where I want ONLY the rids of a set of vertices that match a condition. I was getting negative rids until I ran across this post: 

https://groups.google.com/forum/#!topic/orient-database/EyLm1lVMolI

This worked in 2.1.x but does not anymore. Now `SELECT @rid as rid FROM` whatever returns a list of ODocuments in the rid field of the ResultSet. When I call `resultSet.getString("rid")`, I get the entire ODocument as a String. I cannot imagine this is the intended behavior.

I have not been able to find any documentation in either 2.1 or 2.2 that explains how to do this correctly.

What is the canonical way to get the real rid as a projection and can someone please, please add it to the docs? This appears to be out of date: http://orientdb.com/docs/last/JDBC-Home.html#first-get-a-connection

**Comment by decafmainline** [comment]
I'm working around this right now by removing the projection all together and then using `String rid = resultSet.getString("@rid");` which returns exactly what I'm looking for. I would still like to know what the proper way of doing an rid projection is.

**Comment by lvca** [comment]
Weird, this is to be working. I guess it's an implicit lazy loading done by the JDBC driver. @robfrank can you check this please?

**Comment by robfrank** [comment]
I checked and this  the behaviour is the same from 2.1 to 2.2.x to next 3.0.x . 
But it doesn't depend on the JDBC driver itself. The same happens using the plain OrientDB API:
```
    List<ODocument> docs = db.query(
        new OSQLSynchQuery<ODocument>("SELECT @rid as rid,stringKey, intKey, text, length, date, score FROM Item"));

    for (ODocument doc : docs) {
      System.out.println(doc.field("@rid") + " - " + doc.field("rid"));
    }

``` 

The output will be:
```
#-2:0 - Item#25:0{stringKey:1,score:959,intKey:1,text:OrientDB is... 
#-2:1 - Item#25:1{stringKey:2,score:479,intKey:2,text:OrientDB is... 
#-2:2 - Item#25:2{stringKey:3,score:319,intKey:3,text:OrientDB is... 
#-2:3 - Item#25:3{stringKey:4,score:239,intKey:4,text:OrientDB is...
#-2:4 - Item#25:4{stringKey:5,score:191,intKey:5,text:OrientDB is 
```

The answer on the mailing list you linked shows the same case, I past the code for completeness:
```
String queryText = "SELECT @rid as rid FROM V";
List<ODocument> results = graph.getRawGraph().query(new OSQLSynchQuery<ODocument>(queryText));
for(ODocument result : results) {
   ODocument object = (ODocument) result.field("rid");
   System.out.println(object.getIdentity());
}
```

**Comment by decafmainline** [comment]
The console output remains mostly unchanged, and the Document API output may be the same, but the JDBC output is definitely different.

But, assuming this is intended behavior, I'll go back to my original question. What is the canonical way of retrieving only the rid for a set of vertices using a SQL projection?

**Comment by robfrank** [comment]
The commits are solving the problem and will be included in the next 2.2.12 release and 3.0.0 .

```
    ResultSet rs = stmt.executeQuery("SELECT @rid AS rid, @class, stringKey, intKey, text, length, date FROM Item");

    rs.next();

rs.getString("rid")
```
